File: app/lib/domain/entities/route_point.dart
import 'package:equatable/equatable.dart';

class RoutePoint extends Equatable {
  final double latitude;
  final double longitude;
  final double altitude;
  final double speed; // m/s from GPS sensor
  final DateTime timestamp;

  const RoutePoint({
    required this.latitude,
    required this.longitude,
    required this.altitude,
    required this.speed,
    required this.timestamp,
  });

  @override
  List<Object?> get props => [
        latitude,
        longitude,
        altitude,
        speed,
        timestamp,
      ];
}

File: backend-api/src/DriveJournal.Domain/Entities/RoutePoint.cs
namespace DriveJournal.Domain.Entities;

public class RoutePoint
{
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public double Altitude { get; set; }
    public double Speed { get; set; }
    public DateTime Timestamp { get; set; }
}
